pkg/skbn/abs.go

```diff
@@ -0,0 +1,184 @@
+package skbn
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+var err error
+
+func validateAbsPath(pathSplit []string) error {
+	if len(pathSplit) >= 1 {
+		return nil
+	}
+	return fmt.Errorf("illegal path: %s", filepath.Join(pathSplit...))
+}
+
+func initAbsVariables(split []string) (string, string, string) {
+	account := split[0]
+	container := split[1]
+	path := filepath.Join(split[2:]...)
+
+	return account, container, path
+}
+
+func getNewPipeline() (pipeline.Pipeline, error) {
+	accountName, accountKey := os.Getenv("AZURE_STORAGE_ACCOUNT"), os.Getenv("AZURE_STORAGE_ACCESS_KEY")
+
+	if len(accountName) == 0 || len(accountKey) == 0 {
+		log.Fatal("Either the AZURE_STORAGE_ACCOUNT or AZURE_STORAGE_ACCESS_KEY environment variable is not set")
+	}
+
+	credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+
+	if err != nil {
+		log.Fatal("Invalid credentials with error: " + err.Error())
+	}
+	pl := azblob.NewPipeline(credential, azblob.PipelineOptions{})
+
+	return pl, err
+}
+
+func getServiceURL(pl pipeline.Pipeline, accountName string) (azblob.ServiceURL, error) {
+	URL, err := url.Parse(
+		fmt.Sprintf("https://%s.blob.core.windows.net/", accountName))
+
+	surl := azblob.NewServiceURL(*URL, pl)
+	return surl, err
+}
+
+func getContainerURL(pl pipeline.Pipeline, accountName string, containerName string) (azblob.ContainerURL, error) {
+	URL, err := url.Parse(
+		fmt.Sprintf("https://%s.blob.core.windows.net/%s", accountName, containerName))
+
+	curl := azblob.NewContainerURL(*URL, pl)
+	return curl, err
+}
+
+func getBlobURL(curl azblob.ContainerURL, blob string) (azblob.BlockBlobURL, error) {
+	return curl.NewBlockBlobURL(blob), err
+}
+
+func createContainer(ctx context.Context, pl pipeline.Pipeline, curl azblob.ContainerURL) (*azblob.ContainerCreateResponse, error) {
+	cr, err := curl.Create(ctx, azblob.Metadata{}, azblob.PublicAccessNone)
+	return cr, err
+}
+
+func listContainers(ctx context.Context, surl azblob.ServiceURL) ([]azblob.ContainerItem, error) {
+	lc, err := surl.ListContainersSegment(ctx, azblob.Marker{}, azblob.ListContainersSegmentOptions{})
+	return lc.ContainerItems, err
+}
+
+func containerExists(list []azblob.ContainerItem, containerName string) bool {
+	exists := false
+	for _, v := range list {
+		if containerName == v.Name {
+			exists = true
+		}
+	}
+	return exists
+}
+
+// GetClientToAbs checks the connection to azure blob storage and returns the tested client (pipeline)
+func GetClientToAbs(ctx context.Context, path string) (pipeline.Pipeline, error) {
+	pSplit := strings.Split(path, "/")
+	a, c, _ := initAbsVariables(pSplit)
+	pl, err := getNewPipeline()
```

Comment:
I think we should improve the error handling.
What happens if err != nil here? Or in any other line?
I think that every error should be checked and if err != nil - the flow should stop and the error should be returned to the caller.

Reply:
I will update it. I haven't done it before because some types can't be nil

Reply:
Fixed